pandas/core/indexes/datetimes.py

```diff
@@ -1311,14 +1311,28 @@ def _parsed_string_to_bounds(self, reso, parsed):
         ----------
         reso : Resolution
             Resolution provided by parsed string.
-        parsed : datetime
+        parsed : datetime or object
             Datetime from parsed string.
 
         Returns
         -------
         lower, upper: pd.Timestamp
 
         """
+        parsed = Timestamp(parsed)
+        if self.tz is None:
+            if parsed.tz is None:  # both are naive, nothing to do
+                pass
+            else:  # naive datetime index but label provides timezone
+                warnings.warn("Access naive datetime index with a label "
+                              "containing a timezone, assume UTC")
+                parsed = parsed.tz_convert(utc)
+        else:
+            if parsed.tz is None:  # treat like in same timezone
+                parsed = parsed.tz_localize(self.tz)
```

Comment:
This case already worked before AFAIK, do you know why this is needed? (although the code seems logical)

Reply:
The code is not necessarily needed when it is done somewhere else